File: Backend/OnlyCards/src/main/java/unical/enterpriceapplication/onlycards/application/data/entities/cards/Condition.java
package unical.enterpriceapplication.onlycards.application.data.entities.cards;

import lombok.Getter;

import java.util.Arrays;
import java.util.Map;
import java.util.Optional;
import java.util.stream.Collectors;

@Getter
public enum Condition{
    MINT("mint", "Perfetta"),
    NEAR_MINT("near-mint", "Come nuova"),
    EXCELLENT("excellent", "Leggera Usura"),
    GOOD("good", "Usura Visibile"),
    LIGHT_PLAYED("light-played", "Grave Usura"),
    PLAYED("played", "Danneggiata"),
    POOR("poor", "Distrutta");
    private final String key;
    private final String value;

    Condition(String key, String value) {
        this.key = key;
        this.value = value;
    }

    public static Map<String, String> getConditions() {
        return Arrays.stream(Condition.values())
                .collect(Collectors.toMap(Condition::getKey, Condition::getValue));
    }
    public static Optional<Condition> fromKey(String key) {
        return Arrays.stream(Condition.values())
                .filter(e -> e.getKey().equals(key))
                .findFirst();
    }
}

File: AndroidApp/OnlyCards/app/src/main/java/unical/enterpriceapplication/onlycards/viewmodels/enumData/Condition.kt
package unical.enterpriceapplication.onlycards.viewmodels.enumData

enum class Condition(val key: String, val value: String) {
    MINT("mint", "Mint"),
    NEAR_MINT("near-mint", "Near Mint"),
    EXCELLENT("excellent", "Excellent"),
    GOOD("good", "Good"),
    LIGHT_PLAYED("light-played", "Light Played"),
    PLAYED("played", "Played"),
    POOR("poor", "Poor");

    companion object {
        fun fromString(conditionValue: String): Condition {
            return when (conditionValue) {
                "Mint" -> MINT
                "Near Mint" -> NEAR_MINT
                "Excellent" -> EXCELLENT
                "Good" -> GOOD
                "Light Played" -> LIGHT_PLAYED
                "Played" -> PLAYED
                "Poor" -> POOR
                else -> throw IllegalArgumentException("Invalid condition value")
            }
        }   // Ritorna l'oggetto Condition corrispondente alla stringa passata
        fun getValues(): List<String> {
            return entries.map { it.value }
        }   // Ritorna la lista delle chiavi
    }
}